### PR TITLE
gh-116646: Add PyLong_FileDescriptor_Converter() function

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-116646

----
📚 Documentation preview 📚: https://cpython-previews--116655.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/long.rst [hunk 1]
```diff
@@ -464,3 +464,10 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
 
    Otherwise, the return value is undefined.
 
+.. c:function:: int PyLong_FileDescriptor_Converter(PyObject *p, void *ptr)
+
+   Convert a Python object to a file descript (C ``int`` type).
+   Set `*ptr` and return ``1`` on success. Set an exception and return ``0`` on error.
```
Comment (erlend-aasland) [inline Doc/c-api/long.rst]: I *think* this is the correct markup:

```suggestion
   Set *\*ptr* and return ``1`` on success. Set an exception and return ``0`` on error.
```

**Review by erlend-aasland — COMMENTED** [review]
Consider to add a macro that adds an alias to the previous underscored name.

**Comment by vstinner** [comment]
@erlend-aasland:
> Consider to add a macro that adds an alias to the previous underscored name.

Oh, I didn't remove it on purpose, I just didn't notice the "removal".

Hum, `_PyLong_FileDescriptor_Converter()` is declared in the **internal** `Include/internal/pycore_fileutils.h` header in Python 3.13. Oh, it was declared in ``Include/cpython/fileobject.h`` in Python 3.12.

Ok, I added an alias in `Include/cpython/fileutils.h`: not in the limited C API (`Include/fileutils.h`).

**Comment by erlend-aasland** [comment]
I think you need to update the stable API manifest as well: `Misc/stable_abi.toml`

File: Doc/c-api/long.rst [hunk 2]
```diff
@@ -464,3 +464,10 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
 
    Otherwise, the return value is undefined.
 
+.. c:function:: int PyLong_FileDescriptor_Converter(PyObject *p, void *ptr)
+
+   Convert a Python object to a file descript (C ``int`` type).
+   Set *\*ptr* and return ``1`` on success. Set an exception and return ``0`` on error.
```
Comment (erlend-aasland) [inline Doc/c-api/long.rst]: Style nit (feel free to ignore); I prefer semantic line-breaks for new docs :)

```suggestion
   Set *\*ptr* and return ``1`` on success.
   Set an exception and return ``0`` on error.
```

Comment (vstinner) [reply Doc/c-api/long.rst]: Rendered as HTML, it's the same, no? I applied your suggestion ☺️

Comment (erlend-aasland) [reply Doc/c-api/long.rst]: Yes, it renders the same, but the diff is nicer if we need to amend one of the sentences in the future ;)

**Comment by vstinner** [comment]
See also PR https://github.com/python/cpython/pull/116736

**Comment by vstinner** [comment]
I wrote https://github.com/python/cpython/pull/116769 which replaces this PR.

**Pull request closed without merging**